raiden-ts/src/services/epics.ts

```diff
@@ -1321,7 +1323,22 @@ function parsePfsResponse(
     // decode results and cap also client-side for pfsMaxPaths
     const results = decode(PathResults, data).result.filter((_, idx) => idx < pfsMaxPaths);
     const paths = results.map(({ path, estimated_fee }) => {
-      const fee = estimated_fee.mul(Math.round(pfsSafetyMargin * 1e6)).div(1e6) as Int<32>;
+      let fee;
+      if (estimated_fee.lte(0)) fee = estimated_fee;
+      // add fee margins iff estimated_fee is greater than zero
+      else {
+        const [feeMultiplier, amountMultiplier] =
+          typeof pfsSafetyMargin === 'number'
+            ? [pfsSafetyMargin, 0] // legacy: receive feeMultiplier directly, e.g. 1.1 = +10%
+            : [pfsSafetyMargin[0] + 1, pfsSafetyMargin[1]]; // feeMultiplier = feeMargin% + 100%
+        fee = decode(
+          Int(32),
+          new BN(estimated_fee.toHexString())
+            .times(feeMultiplier)
+            .plus(new BN(value.toHexString()).times(amountMultiplier))
+            .toFixed(0), // fee = estimatedFee * (feeMultiplier) + amount * amountMultiplier
+        );
+      }
```

Comment:
Unfortunately I had to do these refactorings because of CodeClimate complaints. I'll split this part on yet another helper function, since two statements was already too much for this case.

Reply:
Split.